### ISSUE TITLE
v2.15.0: macOS x86_64 binary is actually an arm64 binary

### ISSUE THREAD

**Issue by shenxianpeng** [hadolint/hadolint#1223]
## Summary

The v2.15.0 release ships an **arm64 binary** under the name `hadolint-macos-x86_64`. This means the binary cannot run on Intel (x86_64) Macs.

## Evidence

| File | Actual architecture | Size | SHA256 |
|---|---|---|---|
| `hadolint-macos-arm64` | arm64 | 102,590,520 | `1d701a41d2760ec88f8eff8d93f78b743d6ed57c53146ef274dbeb540816e563` |
| `hadolint-macos-x86_64` | **arm64** (same!) | 102,590,520 | `1d701a41d2760ec88f8eff8d93f78b743d6ed57c53146ef274dbeb540816e563` |

```shell
$ file hadolint-macos-x86_64
hadolint-macos-x86_64: Mach-O 64-bit executable arm64

$ lipo -info hadolint-macos-x86_64
Non-fat file: hadolint-macos-x86_64 is architecture: arm64

$ diff hadolint-macos-arm64 hadolint-macos-x86_64
# (identical - no output)
```

For comparison, v2.14.0's x86_64 binary was correctly built:

```shell
$ file hadolint-v2.14.0-macos-x86_64
hadolint-v2.14.0-macos-x86_64: Mach-O 64-bit executable x86_64
```

## Impact

Running the binary on Intel Macs produces:

```
Bad CPU type in executable
```

## Expected behavior

`hadolint-macos-x86_64` should be a **x86_64** (or universal/fat) binary that runs on Intel Macs.

**Comment by shenxianpeng** [comment]
This is fixed by #1224 — macos x86_64 build now uses `macos-15-intel` runner.
